app/src/main/java/com/example/concordia_campus_guide/Activities/MainActivity.java

```diff
@@ -117,26 +163,29 @@ private void setupSideMenuItemListeners() {
         NavigationView navigationView = findViewById(R.id.side_nav_view);
         if (navigationView != null) {
             navigationView.setNavigationItemSelectedListener(this);
+            setupSwitchesListener(navigationView);
         }
-        setupSwitchesListener(navigationView);
-
     }
 
     private void setupSwitchesListener(NavigationView navigationView) {
-        int[] switchIds = { R.id.nav_translate, R.id.nav_staff, R.id.nav_accessibility};
+        MenuItem translateItem = navigationView.getMenu().findItem(R.id.nav_translate);
+        translationToggle = (CompoundButton) translateItem.getActionView();
+        setupOnChangeListenerForSwitch(translationToggle);
 
-        for(int switchId : switchIds){
-            MenuItem switchItem = navigationView.getMenu().findItem(switchId);
-            CompoundButton switchView = (CompoundButton) MenuItemCompat.getActionView(switchItem);
-            setupOnChangeListenerForSwitch(switchView);
-        }
+        MenuItem staffItem = navigationView.getMenu().findItem(R.id.nav_staff);
+        staffToggle = (CompoundButton) staffItem.getActionView();
+        setupOnChangeListenerForSwitch(staffToggle);
+
+        MenuItem accessibilityItem = navigationView.getMenu().findItem(R.id.nav_accessibility);
+        accessibilityToggle = (CompoundButton) accessibilityItem.getActionView();
+        setupOnChangeListenerForSwitch(accessibilityToggle);
```

Comment:
initially, I did the list and for loop to avoid repeating similar lines for each item we have in the nav view, for now, we have 3 toggles which is not a lot but if we had 10 it would have been really repetitive. You could still have a list of your MenuItems and then loop through it, inside the loop you would get a toggle from the menu item and call its onChangeListener.  This is just an idea I had, let me know if you like it or no. Right now it would help remove 4 duplicate lines, it's minor haha

Reply:
Yes I completely agree about it being redundant! I removed your loop because I add to use the compound buttons somewhere else in the code but you were creating them only in this portion of the code inside the for loop. That was a quick solution tho, I'll look into your suggestion instead.